demo_scripts/moodle/README.md

```diff
@@ -0,0 +1,72 @@
+# Deploy a scalable Moodle stack on Azure
+
+A pre-requisite to going through this tutorial is `az login`. Please run `az login` from the command line and get back to this deployment tutorial. Many commands in this deployment experience require you to specifiy values. In order to get going as quickly as possible we've provided defaults for you.
+
+Here are the defaults that will be used:
+```
+env | grep MOO
+```
+# Creating a local workspace for Moodle ARM and other configuration files
+```
+mkdir -p $MOODLE_AZURE_WORKSPACE/$MOODLE_RG_NAME
+```
+# We will use SSH to access infrastructure post-deployment and the following command will create a Secure Shell (SSH) key-pair to assist for when you're ready to remotely connect to your Moodle deployment via SSH
+```
+if [ ! -f "$MOODLE_SSH_KEY_FILENAME" ]; then ssh-keygen -t rsa -N "" -f $MOODLE_SSH_KEY_FILENAME; fi
+```
+
+# Grab the public part of the SSH key-pair to add to our configuration: this will help with remote connectivity later on
+```
+export SSH_PUBLIC_KEY=$(cat $MOODLE_SSH_KEY_FILENAME.pub)
+```
+
+# Verify public key string before writing to the deployment configuration parameters
+```
+echo $SSH_PUBLIC_KEY
```

Comment:
This still isn't validating anything, it simply an echo to the the shell. It needs a test to ensure that there is actually public key. Anything that requires a human to do something breaks from the design principles of an executable doc.

Something like:

```
#Results

` ` ` 
ssh-rsa AAAAB3NzaC1y...1Eamz/nT4I3 root@localhost
` ` `
```

It won't be very accurate but it will at least guard against an empty value.